### ISSUE TITLE
Calculated pre-devel version is incompatible with the docs

### ISSUE THREAD

**Issue by piotrminkina** [DanySK/git-sensitive-semantic-versioning-gradle-plugin#595]
Hello,

I read from the documentation that:

    "Archeo" versions for the development before initializing git, in the form of 0.1.0-archeo+time
    "Pre-development" versions for the development before marking the first release, in the form of 0.1.0-dev+hash

But in practice this works so that `archeo` is added even if the repository is initialized, has commits and the working tree is or isn't clean.

I applied the plugin in the simplest possible way:

```groovy
plugins {
    id 'org.danilopianini.git-sensitive-semantic-versioning' version '1.1.10'
}

tasks.register('printVersion') {
    doLast {
        logger.lifecycle('{}', project.version)
    }
}

tasks.named('wrapper') {
    distributionType = Wrapper.DistributionType.ALL
    gradleVersion = '8.2.1'
}
```

The last 3 commits from the repo have such hashes:
- d9444d2e2d54781c554bb925750bfb904cdd3a1e (HEAD -> master)
- a5414b8da660f8f565fd5816cc4a033fd42c3c27
- 0f0ddd83d92403e0a88ab81cc1e48d7b58c2c4ea (origin/master)

And the task `printVersion` shows such a version `0.1.0-archeo+d9444d2` instead of something like `0.1.0-dev+d9444d2`.

Regards
Piotr Minkina

**Comment by piotrminkina** [comment]
Also I read from the documentation that:

    "Development" versions for changes over a tag, in the form 0.1.1-dev01-hash, with the number after dev counting the distance in commits since the last tag.

The next versions after the release of `0.1.0` should be `0.1.1-dev01-hash`, `0.1.1-dev02-hash` etc., but instead they are `0.1.0-dev01-hash`, `0.1.0-dev02-hash` etc.

**Comment by DanySK** [comment]
Hi @piotrminkina 
both behaviors are intended, currently.

 "Archeo" versions are meant to support development while there is **no tag**. It cannot show `0.1.0-devXX`, as these versions are meant to indicate a development *successive* to `0.1.0`.

"Dev" versions work similarly to `git describe`: they fetch the last tag and produce a version counting the number of commits.

I can see the logic of your reasoning, though. Given the way semantic versioning 2.0.0 sorts versions, it would make sense to switch.

**Comment by piotrminkina** [comment]
I understand the meaning of "archeo" and I reported a problem that "archeo" is used even when my repository has already been initialised. As I mentioned, this is a behaviour that is not in line with the documentation of your plugin.

I like the counting of commits until the next development version. The problem is that the version from which this counting occurs is not bumped first. As I mentioned, since the release of version `0.1.0` the next one should be something like `0.1.1-dev01+fffffff` or `0.2.0-dev01+fffffff`. It would be nice if an user could choose version part to bump: next major, minor or bugfix.

The current Plugin behaviour for "Development" versions is also not in line with the documentation of your plugin.

Finally, thank you for your work. It is very useful.

**Comment by DanySK** [comment]
Fixed by #603
